members/admin/department_admin.py

```diff
@@ -128,3 +134,102 @@ def waitinglist_count_link(self, item):
+          "Kaptajn"; "Kaptajn-email"; "Kaptajn-telefon";\
+          "Adresse"; "Post#"; "By"; "Region";\
+          "Dato-sidste-forløb";\
+          "Dato-sidste-arrangement";\
+          "Dato-sidste-foreningsmedlemskab";\
+          "Dato-sidste-støttemedlemskab"\n"""
+
+        # There can be multiple departmentleaders (or even none)
+
+        for d in queryset.order_by("name"):
+            info1 = d.union.name + ";"
+            info1 += d.name + ";"
+            if d.created is not None:
+                info1 += d.created.strftime("%Y-%m-%d")
+            info1 += ";"
+            if d.closed_dtm is not None:
+                info1 += d.closed_dtm.strftime("%Y-%m-%d")
+            info1 += ";"
+            info2 = d.address.streetname
+            if d.address.housenumber != "":
+                info2 += " " + d.address.housenumber
+            if d.address.floor != "" or d.address.door != "":
+                info2 += ", "
+            if d.address.floor != "":
+                info2 += d.address.floor + "."
+            if d.address.door != "":
+                info2 += d.address.door
+            info2 += ";"
+            info2 += d.address.zipcode + ";"
+            info2 += d.address.city + ";"
+            info2 += d.address.region + ";"
+
+            leaders = d.department_leaders.all().order_by("name")
+
+            last_activity = (
+                Activity.objects.all()
+                .filter(department=d, activitytype="FORLØB")
+                .order_by("-start_date")
+                .first()
+            )
+            if last_activity is None:
+                info2 += ";"
+            else:
+                info2 += last_activity.start_date.strftime("%Y-%m-%d") + ";"
+
+            last_activity = (
+                Activity.objects.all()
+                .filter(department=d, activitytype="ARRANGEMENT")
+                .order_by("-start_date")
+                .first()
+            )
+            if last_activity is None:
+                info2 += ";"
+            else:
+                info2 += last_activity.start_date.strftime("%Y-%m-%d") + ";"
+
+            last_activity = (
+                Activity.objects.all()
+                .filter(department=d, activitytype="FORENINGSMEDLEMSKAB")
+                .order_by("-start_date")
+                .first()
+            )
+            if last_activity is None:
+                info2 += ";"
+            else:
+                info2 += last_activity.start_date.strftime("%Y-%m-%d") + ";"
+
+            last_activity = (
+                Activity.objects.all()
+                .filter(department=d, activitytype="STØTTEMEDLEMSKAB")
+                .order_by("-start_date")
+                .first()
+            )
+            if last_activity is None:
+                info2 += ";"
+            else:
+                info2 += last_activity.start_date.strftime("%Y-%m-%d") + ";"
```

Comment:
Her ville man godt kunne lave en funktion, som modtager info2 (kan vi få et bedre navn end det?) og aktivitetstype, og så leverer den variablen tilbage med enten et ekstra ; eller info om aktivitetn og ;

Reply:
Måske kan du endda slippe for at definere dem manuelt ved at loope igennem dem fra databasen...